scripts/docker/build_docker.py

```diff
@@ -307,39 +529,66 @@ def __parse_arguments(args_list):
+                                               f'repo(s)')
+    dockers_json_path = os.path.join(Paths.gatk_sv_path, "input_values", "dockers.json")
+    docker_remote_args_group.add_argument('--input-json', type=str, default=dockers_json_path,
+                                          help="Path to dockers.json to use as input")
+    docker_remote_args_group.add_argument('--output-json', type=str, default=dockers_json_path,
+                                          help=f"Path to output updated dockers.json. Set to {Paths.dev_null} to turn "
+                                               "off updates")
+    parser.add_argument("--dry-run", action="store_true",
+                        help="Compute docker images that will be build, but don't actually build or push")
+    parser.add_argument("--skip-dependent-images", action="store_true",
+                        help="Don't build images that depend on targets. Can leave images in an unreproducible state: "
+                             "only use this if you are sure you know what you are doing!")
+    parser.add_argument(
+        '--targets', nargs='*', type=str,
+        help='the sub project docker(s) you want to build (note "all" does not include melt)'
+    )
+    short_git_hash_head = get_command_output("git rev-parse --short HEAD").strip()
+    parser.add_argument('--image-tag', type=str, default=short_git_hash_head,
+                        help='tag to be applied to all images being built')
     parser.add_argument(
+        # flag to turn off git protection (default mode is refusing to build when there are untracked files and/or
+        # uncommitted changes)
         '--disable-git-protect', action='store_true',
         help='disable git check/protect when building from local files (will use uncommited changes to build)'
     )
-    parser.add_argument('--skip-base-image-build', action='store_true',
-                        help='skip rebuild of the target\'s base image(s). Assumes that the base image(s) already '
-                             'exist with same tag.')
     parser.add_argument('--no-force-rebuild', action='store_true',
                         help='Do not rebuild docker images if the exact image and tag already exist.')
     parser.add_argument('--skip-cleanup', action='store_true',
                         help='skip cleanup after successful and unsuccessful build attempts.')
+    parser.add_argument('--base-git-commit', type=str,
+                        help="When auto-specifying build targets, base git commit to check for changes. Can be a SHA or"
+                             "other specifier (e.g. HEAD^)")
+    parser.add_argument('--current-git-commit', type=str,
+                        help="When auto-specifying build targets, current git commit to check for changes. If omitted,"
+                             " use current status of git repo with uncommitted changes. Can be a SHA or other specifier"
+                             " (e.g. HEAD)")
```

Comment:
My personal feeling is that the current action provides defaults that are sensible and abide by the principle of least surprise (i.e. use the current state of the repo). I've provided tools so that someone who wants to do something weird can try to do so. If they do something totally crazy, the only thing that can come of it is that they get a weird set of targets. But I've deliberately left open the possibility to manually specify targets (so that someone who knows what they're doing and wants to do something non-standard can). If I start trying to lock down what people can do fully then I think all that will happen is I'll restrain people doing legitimate work that I didn't anticipate.

Reply:
Sure, we could make sure we're using it the right way.